### PR TITLE
Ignore source-code-blocks while processing comments.

### PR DESCRIPTION
**Pull request by jodyhagins**
Appledoc gives warnings when it encounters parsing issues inside
source code blocks.  In particular, it looks for style markup and
reference links.  These are especially bad within ObjectiveC
source code example blocks because of the ObjectiveC bracket
syntax and the obligatory underscores for category methods on
Cocoa classes.

This is similar to bug #72, but not exactly.  I noticed that in
another similar report, these issues are gone in version 3, so I
didn't bother trying to fix this completely.

Basically, the code intercepts the markdown processing, and looks
for any sections within documentation blocks that are marked with
either @code/@endcode of the ~~~/~~~ and ```/``` markdown syntax.

If it finds those blocks, it will output them as markup (making
sure to use the backtick format), and skip any further processing
for that block of text.

### CODE REVIEW

**Comment by tomaz** [comment]
Thanks, merged.
